### DISCUSSION TITLE
mqtt-over-ws-client - How to add user/password to mqtt

### DISCUSSION THREAD

**Discussion by ondonadas** [cesanta/mongoose#2488] [Ask questions, get answers] [answered]
Hi group
I'm trying to understand how to modify [mqtt-over-ws-client](https://github.com/cesanta/mongoose/tree/master/examples/mqtt-over-ws-client) example in order to support user/password authentication.

I guess that something like this should work but, actually, it doesn't work.

>   struct mg_mqtt_opts opts = {
                              .client_id = mg_str("arkanoide"),
                              .user = mg_str(s_user),
                              .pass = mg_str(s_pass),
    };
>
>    size_t len = c->send.len;
    mg_mqtt_login(c, &opts);
    mg_ws_wrap(c, c->send.len - len, WEBSOCKET_OP_BINARY);

Somebody with experience with that can help me.

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation),  [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.

https://mongoose.ws/documentation/#mqtt-1

You seem to be doing what needs to be done, then, "doesn't work" is subjective, localized, and doesn't add anything that can help us solve your problem.
Please sniff your network (Wireshark), see what is going on, and then if necessary don't hesitate to open an issue, fill the issue template, and provide context and network _captures_ so we can help you fix the issue.
